lib/hanami/cli/generator.rb

```diff
@@ -0,0 +1,146 @@
+require 'hanami/utils/files'
+require 'erb'
+
+module Hanami
+  class CLI
+    # A class to help generate files
+    #
+    # @since 0.1.0
+    class Generator
+      # @since x.x.x
+      # @api public
+      def initialize(out: $stdout, files: Utils::Files, templates_dir: __dir__)
```

Comment:
I'm no sure if `__dir__` is a good default here. The other option is that we let it be `nil`, then raise a helpful error if someone tries to use `create` or `copy` without passing this argument in.

I'm also not sure `templates_dir` is the right name for this variable. For `create`, it's the right name, but for `copy`, the files are necessarily templates. For example, in `hanami/hanami`, we copy over a `favicon.ico`, which isn't a template at all. Maybe `root_dir` would be better?

Reply:
@cllns The problem with `root_dir` it can be confused with the root of a project (eg. `Hanami.root`).

Reply:
:+1: It's already been changed to `templates_dir`